alpine/latest/Dockerfile

```diff
@@ -0,0 +1,91 @@
+###############################################################
+#
+# Copyright (C) 2019 James Fuller <jim.fuller@webcomposite.com>
+#
+# SPDX-License-Identifier: MIT
+#
+# pinning to Alpine 3.10.2 as base image
+###############################################################
+FROM registry.hub.docker.com/library/alpine:3.10.2 AS builder
+
+###############################################################
+# set build args
+###############################################################
+ARG CURL_RELEASE_TAG=latest
+ARG CURL_GIT_REPO=https://github.com/curl/curl.git
+ARG CURL_CONFIGURE_OPTION
+ARG LABEL_VERSION=1.0.0
+ARG LABEL_NAME=curl
+ARG LABEL_DESC=curl
+
+###############################################################
+# build curl
+###############################################################
+# install deps and use latest curl release source
+RUN \
+    apk add --no-cache \
+       autoconf    \
+       automake    \
+       build-base  \
+       curl-dev    \
+       groff       \
+       libtool     \
+    && \
+    rm -rf "/var/cache/apk/"*
+
+RUN mkdir /src
+COPY "curl"  "/src/curl"
+WORKDIR "/src/curl"
+
+RUN ./buildconf && \
+    autoreconf -vif && \
+    ./configure ${CURL_CONFIGURE_OPTION} &&\
```

Comment:
Just wondering what the benefit is of using a variable here rather than defining the arguments here.

Reply:
we may want to build different images with different configs (eg. all options, etc) ... though admittedly that may mean needing to add in deps to support. leaving it in as it does no harm.